lib/barbeque/execution_log.rb

```diff
@@ -79,20 +91,33 @@ def legacy_s3_key_for(execution)
     def get(execution, filename)
       s3_object = ExecutionLog.s3_client.get_object(
         bucket: s3_bucket_name,
-        key: s3_key_for(execution, filename),
+        key: s3_key_for_execution(execution, filename),
       )
       s3_object.body.read
     rescue Aws::S3::Errors::NoSuchKey
       nil
     end
 
+    # @param [String] application Application name
+    # @param [String] job Job name
+    # @param [String] message_id Message ID generated by SQS
+    # @param [String] filename
+    # @param [String] content
+    def put(application, job, message_id, filename, content)
```

Comment:
I am confused a little bit about `#put_with_execution` parameters is symmetrical with`#get`.
How about changing the `#get` method name to `#get_with_execution`?
Or changing parameters to `def put(s3_key, content)`  as a private method and change`#get` in the same way?  It is IMO.